bundles/org.eclipse.core.jobs/src/org/eclipse/core/internal/jobs/JobManager.java

```diff
@@ -1100,19 +1101,25 @@ boolean join(InternalJobGroup jobGroup, long timeout, IProgressMonitor monitor)
 	}
 
 	/**
-	 * Returns a non-null progress monitor instance.  If the monitor is null,
-	 * returns the default monitor supplied by the progress provider, or a
-	 * NullProgressMonitor if no default monitor is available.
+	 * Returns a non-null progress monitor instance. If a progress provider is set,
+	 * it is asked to provide a wrapped instance of the given <code>monitor</code>
+	 * parameter. If no progress provider is set and <code>monitor</code> is null, a
+	 * NullProgressMonitor is returned in all other cases the given
+	 * <code>monitor</code> instance.
+	 *
+	 * @param monitor the monitor instance (could be null) for further usage
+	 * @return a non-null progress monitor instance.
 	 */
 	private IProgressMonitor monitorFor(IProgressMonitor monitor) {
-		if (monitor == null || (monitor instanceof NullProgressMonitor)) {
-			if (progressProvider != null) {
-				try {
-					monitor = progressProvider.getDefaultMonitor();
-				} catch (Exception e) {
-					String msg = NLS.bind(JobMessages.meta_pluginProblems, JobManager.PI_JOBS);
-					RuntimeLog.log(new Status(IStatus.ERROR, JobManager.PI_JOBS, JobManager.PLUGIN_ERROR, msg, e));
+		if (progressProvider != null) {
+			try {
+				IProgressMonitor monitorFor = progressProvider.monitorFor(monitor);
+				if (monitorFor == null) {
+					throw new IllegalStateException("ProgressProvider " + progressProvider + " returned null!"); //$NON-NLS-1$ //$NON-NLS-2$
 				}
+			} catch (Exception e) {
+				String msg = NLS.bind(JobMessages.meta_pluginProblems, JobManager.PI_JOBS);
+				RuntimeLog.log(new Status(IStatus.ERROR, JobManager.PI_JOBS, JobManager.PLUGIN_ERROR, msg, e));
 			}
 		}
 
```

Comment:
Github review is clearly by far worse compared to gerrit.
How stupid is to limit comments to the changed code area only?
After refactoring the change got a critical bug - the `monitorFor` is not used at all!
See line 1126 / 1128

Reply:
BTW this means, the tests were not complete - please update the test too.

Reply:
> How stupid is to limit comments to the changed code area only?

Its hard to tell if there is a way if one don't know what you have tried, you can comment on any file / line that was changed, if you think there are issues in other parts (that are not changed) you can copy the "permalink" to reference that code part in the PR or better open a new issue.

> After refactoring the change got a critical bug - the `monitorFor` is not used at all!

Sorry I don't understand can you explain?

These are the locations where it is used (but it has not changed):
https://github.com/eclipse-platform/eclipse.platform.runtime/blob/c3f90252e218436750bba5ae41f26bb0c57b52a0/bundles/org.eclipse.core.jobs/src/org/eclipse/core/internal/jobs/JobManager.java#L316

Reply:
Please open code in IDE and check which value `monitorFor()` returns. And no, from the review editor I can't even copy the reference link to the line I want.

Reply:
Click on the `...` and choose "View File" then you can select a line and choose:

![grafik](https://user-images.githubusercontent.com/1331477/165701345-dcc1ec9e-1996-427c-9199-c9862837c38e.png)

if you have the link you can even add Lxxx-Lyyy to reference more than one line.

Reply:
But I already see what you mean ... just in case you have a need for that some time later :-)

Reply:
> After refactoring the change got a critical bug - the `monitorFor` is not used at all!

Done



> BTW this means, the tests were not complete - please update the test too.

There is actually no way to check for this without rely on internal implementation details.

Reply:
> Click on the `...` and choose "View File" then you can select a line and choose:

Sure, I'm aware about that, but this forces me to leave the review context, which is bad.

It turns out more and more, github is basically about code sharing and less about code review. Serious review is not possible here, for example diffing previous commits on same PR with actual state is not possible (in gerrit you can compare patch set 21 with patch set 42 in a second). It is just unusable for me also because of the wrong way to display code in the review etc, so at the end I'm doing most of the reviews now in the IDE and comment in browser, which is making me unhappy and less productive compared to gerrit.

Reply:
It depend,y if you simply add each commit instead of force-push (as I did here) you get several "patch-sets" as well, but everyone seems so prefer something else :-)